classes/migration/upgrade/v3_4_0/PKPI7014_DoiMigration.inc.php

```diff
@@ -0,0 +1,394 @@
+                    return $carry;
+            }
+        }, $data);
+
+        // Prepare insert statements
+        $insertData = [];
+        foreach ($data->enabledDois as $item) {
+            array_push($insertData, $item);
+        }
+        foreach ($data->doiCreationTime as $item) {
+            array_push($insertData, $item);
+        }
+        foreach ($data->enabledDoiTypes as $item) {
+            $item['setting_value'] = json_encode($item['setting_value']);
+            array_push($insertData, $item);
+        }
+        foreach ($data->doiPrefix as $item) {
+            array_push($insertData, $item);
+        }
+        foreach ($data->doiSuffixType as $item) {
+            array_push($insertData, $item);
+        }
+        foreach ($data->doiPublicationSuffixPattern as $item) {
+            array_push($insertData, $item);
+        }
+        foreach ($data->doiRepresentationSuffixPattern as $item) {
+            array_push($insertData, $item);
+        }
+
+        $insertData = $this->prepareSuffixPatternsForInsert($data, $insertData);
+
+        DB::table($this->getContextSettingsTable())->insert($insertData);
+
+        // Add minimum required DOI settings to context if DOI plugin not previously enabled
+        $missingDoiSettingsInsertStatement = DB::table($this->getContextTable())
+            ->select($this->getContextIdColumn())
+            ->whereNotIn($this->getContextIdColumn(), function (Builder $q) {
+                $q->select($this->getContextIdColumn())
+                    ->from($this->getContextSettingsTable())
+                    ->where('setting_name', '=', 'enableDois');
+            })
+            ->get()
+            ->reduce(function ($carry, $item) {
+                $carry[] = [
+                    $this->getContextIdColumn() => $item->{$this->getContextIdColumn()},
+                    'setting_name' => 'enableDois',
+                    'setting_value' => 0,
+                ];
+                $carry[] = [
+                    $this->getContextIdColumn() => $item->{$this->getContextIdColumn()},
+                    'setting_name' => 'doiCreationTime',
+                    'setting_value' => 'copyEditCreationTime'
+                ];
+                $carry[] = [
+                    $this->getContextIdColumn() => $item->{$this->getContextIdColumn()},
+                    'setting_name' => 'useDefaultDoiSuffix',
+                    'setting_value' => 1
+                ];
+                return $carry;
+            }, []);
+
+        DB::table($this->getContextSettingsTable())->insert($missingDoiSettingsInsertStatement);
+
+        // Cleanup old DOI plugin settings
+        DB::table('plugin_settings')
+            ->where('plugin_name', '=', 'doipubidplugin')
+            ->delete();
+        DB::table('versions')
+            ->where('product_type', '=', 'plugins.pubIds')
+            ->where('product', '=', 'doi')
+            ->update(['current' => 0]);
```

Comment:
Is there a reason this is being marked "not current" rather than deleted as above?

Reply:
(I guess to preserve the history -- but because this plugin always shipped with OJS, there's nothing gained by keeping it, I don't think. I think the `current` flag is only used/considered for the application, not plugins.)

Reply:
Yep, it was just to preserve the history, but all the better if it can be deleted.